test/SlowTests/Issues/RavenDB_14235.cs

```diff
@@ -0,0 +1,256 @@
+
+                Assert.Equal(str, u.WeirdName);
+                Assert.Equal(str, id);
+                Assert.Equal(str, collection);
+            }
+        }
+
+        [Fact]
+        public async Task ShouldThrowOnNotAwaitedAsyncMethods()
+        {
+            const string name = "1";
+
+            var exceptionsCounter = 0;
+            var tryCounter = 0;
+            using var store = GetDocumentStore();
+            try
+            {
+                tryCounter++;
+                using var session = store.OpenAsyncSession();
+                session.LoadAsync<User>(name);
+            }
+            catch (Exception e)
+            {
+                Assert.Equal(typeof(InvalidOperationException), e.GetType());
+                exceptionsCounter++;
+            }
+
+            try
+            {
+                tryCounter++;
+                using var session = store.OpenAsyncSession();
+                ;
+                await session.StoreAsync(new User());
+                session.SaveChangesAsync();
+            }
+            catch (Exception e)
+            {
+                Assert.Equal(typeof(InvalidOperationException), e.GetType());
+                exceptionsCounter++;
+            }
+
+            try
+            {
+                tryCounter++;
+                using var session = store.OpenAsyncSession();
+                session.Query<User>().ToListAsync();
+            }
+            catch (Exception e)
+            {
+                Assert.Equal(typeof(InvalidOperationException), e.GetType());
+                exceptionsCounter++;
+            }
+
+            try
+            {
+                tryCounter++;
+                using var session = store.OpenAsyncSession();
+                session.Advanced.Attachments.GetAsync(name, name);
+            }
+            catch (Exception e)
+            {
+                Assert.Equal(typeof(InvalidOperationException), e.GetType());
+                exceptionsCounter++;
+            }
+
+            try
+            {
+                tryCounter++;
+                using var session = store.OpenAsyncSession();
+                session.Advanced.ClusterTransaction.GetCompareExchangeValueAsync<object>(name);
+            }
+            catch (Exception e)
+            {
+                Assert.Equal(typeof(InvalidOperationException), e.GetType());
+                exceptionsCounter++;
+            }
+
+            try
+            {
+                tryCounter++;
+                using var session = store.OpenAsyncSession();
+                var lazy = session.Advanced.Lazily.LoadAsync<User>(new[] { name, name, name, name });
+                lazy.Value.Start();
+            }
+            catch (Exception e)
+            {
+                Assert.Equal(typeof(InvalidOperationException), e.GetType());
+                exceptionsCounter++;
+            }
+
+            Assert.Equal(tryCounter, exceptionsCounter);
```

Comment:
This will be racy, high chance IMO that some of those will finish and not throw.

Reply:
done